bin/init_exercise.py

```diff
@@ -0,0 +1,411 @@
+#!/usr/bin/env python
+"""
+Script to initialize an exercise for Exercism's Rust track
+
+Why a Python script in the Rust track repo? Distribution.
+A rust program would either need to be precompiled for various
+platforms, and available for download (in which case it wouldn't
+conveniently work in the repository), or would need to be included
+as a sub-crate and compiled locally. A python script can simply
+be a single file and depend on the user's system Python, if desired.
+
+This module requires Python3.5 or newer
+"""
+from __future__ import print_function
+
+try:
+    import collections
+    import json
+    import os
+    import shlex
+    import subprocess
+    import string
+    import sys
+    import urllib.request
+
+    from contextlib import contextmanager
+    from datetime import datetime
+    from uuid import uuid4
+except ImportError:
+    print("This script requires Python 3.5 or higher", file=sys.stderr)
+    # exiting like this isn't great for library use, but at least it's a quick fail
+    sys.exit(1)
+
+# check version info
+if sys.version_info[0] != 3 or sys.version_info[1] < 5:
+    print("This script requires Python 3.5 or higher", file=sys.stderr)
+    # exiting like this isn't great for library use, but at least it's a quick fail
+    sys.exit(1)
+
+
+def output_of(cmd, check_returncode=True):
+    "Return the stdout of the given command"
+    sp = subprocess.run(shlex.split(cmd),
+                        stdout=subprocess.PIPE,
+                        universal_newlines=True)
+    if check_returncode:
+        sp.check_returncode()
+    return sp.stdout.strip()
+
+
+REPO_ROOT = output_of('git rev-parse --show-toplevel')
+EXERCISES = os.path.join(REPO_ROOT, 'exercises')
+ITEM_NAME_CHARS = {c for c in string.ascii_lowercase + string.digits + '_'}
+VERSION = "0.1.0"
+
+
+def to_item_name(description):
+    "Produce a valid rust item name from arbitrary inputs"
+    item = description.lower().replace(' ', '_')
+    item = [c for c in item if c in ITEM_NAME_CHARS]
+    while len(item) > 0 and item[0] in string.digits:
+        item = item[1:]
+    if len(item) == 0:
+        raise ValueError("Could not produce an item name from " + description)
+    return item
+
+
+def url_for(name, file):
+    return "https://raw.githubusercontent.com/exercism/problem-specifications/master/exercises/{name}/{file}".format(
+        name=name, file=file,
+    )
+
+
+def get_problem_specification(name):
+    """
+    Try to get problem specifications for the exercise of the given name.
+
+    If the problem specifications repo doesn't exist, returns None.

```

Comment:
I'm not convinced the condition is true. I think it's "if the specification does not exist in the problem specifications repo", rather than whether the repo doesn't exist

Reply:
You're right, that's a better statement. Both are true, however.